backend/tests/integration/records/api/test_api_literature.py

```diff
@@ -1781,3 +1781,218 @@ def test_adding_files_with_s3_url_but_wrong_key(
 
         assert record["figures"] == expected_figures
         assert record["documents"] == expected_documents
+
+
+def test_creating_record_updates_entries_in_authors_records_table(
+    base_app, db, create_record
+):
+    expected_authors_entries_count = 5
+    expected_table_entries = [
+        "K.Janeway.1;INSPIRE BAI",
+        "INSPIRE-12345678;INSPIRE ID",
+        "J.T.Kirk.1;INSPIRE BAI",
+        "collaboration_1;collaboration",
+        "collaboration_2;collaboration",
+    ]
+    data = {
+        "authors": [
+            {
+                "full_name": "Kathryn Janeway",
+                "ids": [
+                    {"value": "K.Janeway.1", "schema": "INSPIRE BAI"},
+                    {"value": "INSPIRE-12345678", "schema": "INSPIRE ID"},
+                ],
+            },
+            {
+                "full_name": 'James Tiberius "Jim" Kirk',
+                "ids": [{"value": "J.T.Kirk.1", "schema": "INSPIRE BAI"}],
+            },
+        ],
+        "collaborations": [{"value": "collaboration_1"}, {"value": "collaboration_2"}],
+    }
+
+    record = create_record("lit", data=data)
+
+    authors_records_entries = RecordsAuthors.query.filter_by(record_id=record.id).all()
+    assert len(authors_records_entries) == expected_authors_entries_count
+    table_entries = [
+        f"{entry.author_id};{entry.id_type}" for entry in authors_records_entries
+    ]
+    assert sorted(table_entries) == sorted(expected_table_entries)
```

Comment:
then you can simply do
```suggestion
    table_entries = {
        (entry.author_id, entry.id_type) for entry in authors_records_entries
    }
    assert table_entries == expected_table_entries
```

Reply:
I have no idea why I did this weird creation with strings at first... 
:+1: